src/Compilers/CSharp/Portable/Operations/CSharpConversionExpression.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using Microsoft.CodeAnalysis.Semantics;
+
+namespace Microsoft.CodeAnalysis.CSharp
+{
+    internal abstract class BaseCSharpConversionExpression : BaseConversionExpression
+    {
+        protected BaseCSharpConversionExpression(Conversion conversion, bool isExplicitInCode, bool throwsExceptionOnFailure, bool isChecked, SyntaxNode syntax, ITypeSymbol type, Optional<object> constantValue) :
+            base(conversion, isExplicitInCode, throwsExceptionOnFailure, isChecked, syntax, type, constantValue)
+        {
+            ConversionInternal = conversion;
+        }
+
+        internal Conversion ConversionInternal { get; }
+
+        public override string LanguageName => LanguageNames.CSharp;
```

Comment:
1. should be Language not LanguageName.
2. I don't think we need this. *or* we should move this to the root IOperation.  It would fit there given that we have ISymbol.Language and SyntaxNode.Language.   If you want we can ratify this at the design meeting tomorrow.